qlty-check/src/llm/fixer.rs

```diff
@@ -30,20 +32,33 @@ pub struct Fixer {
     progress: Progress,
     staging_area: StagingArea,
     r#unsafe: bool,
-    attempts_per_file: Arc<Mutex<HashMap<Option<String>, AtomicUsize>>>,
+    attempts_per_file: Arc<Mutex<HashMap<String, AtomicUsize>>>,
     total_attempts: Arc<AtomicUsize>,
 }
 
 impl IssueTransformer for Fixer {
-    fn transform(&self, mut issue: Issue) -> Option<Issue> {
-        if !self.reached_max_fixes(&issue) {
-            issue = self
-                .fix_issue(&issue)
-                .inspect(|issue| self.update_max_fixes(issue))
-                .unwrap_or(issue);
-        }
-
-        Some(issue)
+    fn transform_batch(&self, issues: Vec<Issue>) -> Vec<Issue> {
+        let no_path_issues = issues
```

Comment:
Issues without a path are possible but quite rare. So this block of code to separate them out stood out to me. Could we handle the issues without paths in the main iterator instead to avoid needing to keep track of two vectors?

Reply:
Yea I like this idea.